components/TextFields/MDCTextInputBehavior.h

```diff
@@ -0,0 +1,104 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights
+ Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@class MDCTextInput;
+
+/**
+ Material Design themed states for textInputs. The logic for 'automagic' error states changes:
+ underline color, underline text color.
+ https://www.google.com/design/spec/components/text-fields.html#text-fields-single-line-text-field
+ */
+@interface MDCTextInputBehavior : NSObject
+
+@property(nonatomic, nonnull, strong) id<MDCTextInput> textInput NS_SWIFT_NAME(input);
```

Comment:
While I do think some of this would be repeated globally, the best way to do that would be UIAppearance (which I just realized could be added also to underlineViewMode.)

If we isolated behavior to handling just errors, then yes it would make sense to have 1 that is used as needed.

But I believe we can move more behavior into this: things that would take advantage of it as a state manager.